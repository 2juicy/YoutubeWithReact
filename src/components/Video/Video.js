import React from "react";
import "./Video.scss";

export default function Video({ id }) {
  if (!id) {
    return null;
  }

  return (
    <div className="video-container">
      <div className="video">
        <iframe
          className="video-player"
          src={`https://www.youtube.com/embed/${id}?autoplay=1`}
          frameBorder="0"
          title="video"
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen
        ></iframe>
      </div>
    </div>
  );
}
