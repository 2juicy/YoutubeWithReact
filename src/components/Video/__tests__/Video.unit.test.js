import React from "react";
import { shallow } from "enzyme";
import Video from "../Video";

describe("Video", () => {
  test("renders with video", () => {
    const wrapper = shallow(<Video id="0siAhk2DlgA" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders null, no id", () => {
    const wrapper = shallow(<Video />);
    expect(wrapper).toMatchSnapshot();
  });
});
