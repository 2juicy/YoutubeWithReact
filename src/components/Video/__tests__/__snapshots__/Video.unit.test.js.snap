// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders null, no id 1`] = `""`;

exports[`Video renders with video 1`] = `
<div
  className="video-container"
>
  <div
    className="video"
  >
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowfullscreen={true}
      className="video-player"
      frameborder="0"
      src="https://www.youtube.com/embed/0siAhk2DlgA?autoplay=1"
      title="video"
    />
  </div>
</div>
`;
